### DISCUSSION TITLE
[License] Can I just re-distribute the pyodide-core files inside my package?

### DISCUSSION THREAD

**Discussion by salazarbarrera** [pyodide/pyodide#4317] [Q&A]
I'm a bit confused by how the Mozilla license operates here.

You distribute a separate "core" package [containing the minimal set of files](https://pyodide.org/en/stable/usage/downloading-and-deploying.html) needed to start Pyodide with every release. Since this is a minimal package, it doesn't contain a License file, and while the js files are mostly readable, some of the other files are binaries.

I want to eventually release a zip file containing a fully client-side web app ready to be locally uncompressed in a simple web server. Should I add the _whole_ source code to my release package if I wanted to include a local copy of core files in order to comply with the MPL? If not, should I add a full copy of the source code in my repo?

I know this problem could be totally avoided by using npm (letting the job of downloading, building and setting up a node server to each developer), but given that it is not necessary for my use-case, I want it to run as much as "out of the box" as possible.

So: Can I just include a pyodide-core directory along with my index.html?

**Comment by hoodmane** [comment]
I talked to some lawyers about the MPL2 recently. MPL2 says if you use artifacts built from a modified or unmodified copy of our source files, you must provide a reference to the version of the source code used to built the artifacts. You don't need a copy of the Pyodide source code since you haven't modified it. I think it would be good enough to put a file in the folder saying "this is the Pyodide distro" and linking back to this repo and the releases link you downloaded it from.

The lawyers recommended that if you have a commercial product that uses Pyodide there should be a credits banner with a link to the source repo somewhere the users will see. But this seems like a hassle, and it is not clear to me that this "recommendation" is legally required...

The point of making github releases is so that people can use them. We don't want the license to be a usability problem, so probably we should be adding a file linking back to the Pyodide repo to our releases so that people won't be confused.

It doesn't sound like you are postprocessing the files at all, but as I understand it MPL2 does not add any additional constraints if you do postprocessing. For example, if you take the built artifacts and injected extra code into them, webpacked them together with other files, etc. My belief is that the code that does this postprocessing need not be open source.

**Reply by salazarbarrera** [reply]
That makes sense! Thanks!
